src/logic/wallets/utils/network.ts

```diff
@@ -81,8 +81,8 @@ export const switchNetwork = async (wallet: Wallet, chainId: ETHEREUM_NETWORK):
 
 export const shouldSwitchNetwork = (wallet = onboard().getState()?.wallet): boolean => {
   const desiredNetwork = getNetworkId()
-  const currentNetwork = wallet?.provider?.networkVersion?.toString()
-  return currentNetwork ? desiredNetwork !== currentNetwork : false
+  const currentNetwork = wallet?.provider?.networkVersion
+  return currentNetwork ? desiredNetwork !== currentNetwork.toString() : false
```

Comment:
I think we lost a `?` here. Or is it always a string?

Reply:
We are already checking that `currentNetwork` exists before converting to a string in the return statement, if not we return false